lizard_connector/connector.py

```diff
@@ -27,34 +31,52 @@
 
 import lizard_connector.queries
 
+ASYNC_POLL_TIME = 1
+ASYNC_POLL_TIME_INCREASE = 1.5
+
 
 class LizardApiTooManyResults(Exception):
     pass
 
 
+class LizardApiAsyncTaskFailure(Exception):
+    pass
+
+
 class InvalidUrlError(Exception):
     pass
 
 
+def no_op(*args, **kwargs):
+    pass
+
+
+def save_to_json(result, file_base="api_result"):
+    """
+    Saves a result to json with a timestamp in milliseconds.
+
+    Args:
+        result (list|dict): a json dumpable object to save to file.
+        file_base (str): filename base. Can contain a relative or absolute
+                         path.
+    """
+    filename = "{}_{}.json".format(file_base, str(int(time.time() * 1000)))
+    with open(filename, 'w') as json_filehandler:
+        json.dump(result, json_filehandler)
+
+
 class Connector(object):
 
-    def __init__(self, max_results=1000, username=None, password=None,
-                 all_pages=True):
+    def __init__(self, max_results=1000, username=None, password=None):
```

Comment:
Actually, that is the intended behaviour. You don't want people to request more than a thousand synchronously at once. The error message should be more clear though. Furthermore, you'ld want the max results to differ between async and sync.